src/prefect/orion/models/block_documents.py

```diff
@@ -492,11 +492,36 @@ async def update_block_document(
             new_block_document_references,
         ) = _separate_block_references_from_data(update_values["data"])
 
-        # encrypt the data
+        # encrypt the data and write updated data to the block document
         await current_block_document.encrypt_data(
             session=session, data=block_document_data_without_refs
         )
 
+        # `proposed_block_schema` is always the same as the schema on the client-side
+        # Block class that is calling `save`, which may or may not be the same schema
+        # as the one on the saved block document
+        proposed_block_schema_id = block_document.block_schema_id
+
+        # if a new schema is proposed, update the block schema id for the block document
+        if proposed_block_schema_id != current_block_document.block_schema_id:
```

Comment:
We need to check to make sure a new block schema ID was provided. I'd recommend reverting [13c5f52](https://github.com/PrefectHQ/prefect/pull/8056/commits/13c5f527894bdc878e4760c064fd9db2e85fadb7) since it shouldn't be necessary if we're making this check.
```suggestion
        if proposed_block_schema_id != current_block_document.block_schema_id and proposed_block_schema_id is not None:
```